src/sklearn_raster/estimator.py

```diff
@@ -142,61 +143,63 @@ def predict(
         allow_cast: bool = False,
         check_output_for_nodata: bool = True,
         **predict_kwargs,
-    ) -> ImageType:
+    ) -> FeatureType:
         """
-        Predict target(s) for X_image.
+        Predict target(s) for n-dimensional X features.
 
         Parameters
         ----------
-        X_image : Numpy or Xarray image with 3 dimensions (y, x, band)
-            The input image. Features in the band dimension should correspond with the
-            features used to fit the estimator.
+        X : Numpy or Xarray features
+            The n-dimensional input features. Array types should be in the shape
+            (features, ...) while xr.Dataset should include features as variables.
+            Features should correspond with those used to fit the estimator.
         skip_nodata : bool, default=False
             If True, NoData and NaN values will be skipped during prediction. This
-            speeds up processing of partially masked images, but may be incompatible if
+            speeds up processing of partially masked arrays, but may be incompatible if
             estimators expect a consistent number of input samples.
         nodata_input : float or sequence of floats, optional
-            NoData values other than NaN to mask in the output image. A single value
-            will be broadcast to all bands while sequences of values will be assigned
-            band-wise. If None, values will be inferred if possible based on image
-            metadata.
+            NoData values other than NaN to mask in the output array. A single value
+            will be broadcast to all features while sequences of values will be assigned
+            feature-wise. If None, values will be inferred if possible based on
+            available metadata.
         nodata_output : float or int, default np.nan
-            NoData pixels in the input features will be replaced with this value in the
+            NoData in the input features will be replaced with this value in the
```

Comment:
Perhaps consider changing this to `NoData samples`?  For me, "samples" is a good abstraction for "pixels".